x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java

```diff
@@ -278,6 +283,28 @@ static ShutdownShardMigrationStatus shardMigrationStatus(
                 }
                 return hasShardCopyOnOtherNode == false;
             })
+            // If ILM is shinking the index this shard is part of, it'll look like it's unmovable, but we can just wait for ILM to finish
+            .filter(pair -> {
+                if (OperationMode.STOPPED.equals(currentILMMode(currentState)) == false) {
+                    LifecycleExecutionState ilmState = currentState.metadata().index(pair.v1().index()).getLifecycleExecutionState();
+                    // Specifically, if 1) ILM is running, 2) ILM is currently shrinking the index this shard is part of, and 3) it hasn't
+                    // errored out, we can disregard this shard under the assumption that ILM will get it movable eventually
+                    boolean ilmWillMoveShardEventually = ShrinkAction.NAME.equals(ilmState.action())
+                        && ErrorStep.NAME.equals(ilmState.step()) == false;
```

Comment:
Actually it appears that we always build it with `LifecycleExecutionState.EMPTY_STATE` so it's non-nullable, but should we be paranoid here regardless?

Reply:
As an alternative, could we annotate `getLifecycleExecutionState()` and the IndexMetadata builder with NonNull or similar? (Is that a convention in the codebase? Sorry, new here and still learning :) ) to "guarantee" that it will not be null?
One good thing about using NullObjects and similar is to not have to check for null, I would avoid it if we can.

Reply:
We have an `@Nullable` annotation, but not an `@NotNull` or similar. I'm not sure why, as the positive version is less useful if it's used inconsistently. I'll add a null check here just in case, easy enough.